### ISSUE TITLE
Skype for Business bridge

### ISSUE THREAD

**Issue by homebeach** [turt2live/matrix-wishlist#221]
A bridge between Matrix and [Skype for business](products.office.com/skype-for-business). Here is [a documentation of SDK](https://docs.microsoft.com/en-us/skype-sdk/appsdk/skypeappsdk).

**Comment by homebeach** [comment]
There exist [XMPP gateway](https://en.wikipedia.org/wiki/Skype_for_Business_Server#XMPP) for Skype for Business.

**Comment by Mikaela** [comment]
I am not sure if there is any point in this as Microsoft seems to be pushing/hoping people go to Microsoft Teams (https://github.com/turt2live/matrix-wishlist/issues/16) at least in the long term.

* https://docs.microsoft.com/en-us/MicrosoftTeams/faq-journey
* https://docs.microsoft.com/en-us/MicrosoftTeams/teams-and-skypeforbusiness-coexistence-and-interoperability

Is anyone actually using Skype for Business and wishing to see a Matrix bridge?

**Comment by turt2live** [comment]
I know quite a few people who would love this kind of bridge, particularly while they're looking to switch to another platform.

**Comment by homebeach** [comment]
Here is the [protocol documentation](https://msdn.microsoft.com/en-us/library/cc339480(v=office.12).aspx).

**Comment by homebeach** [comment]
[Sameroom](https://sameroom.io) seems to offer bridging for Skype for Business.

**Comment by homebeach** [comment]
According to [Wiki](https://wiki.bitlbee.org/), BitBlee seems to support Skype for Business via [SIPE](https://wiki.bitlbee.org/HowtoSIPE).

**Comment by bes1002t** [comment]
As far as I know skype for business will be completely replaced by MS teams. So pushing a teams bridge forward would make more sense.

**Comment by andreashaerter** [comment]
@bes1002t You are right in general, but there are big institutions on Skype for Business on-prem yet and they still get support by Microsoft. Many of them are not able to migrate to MS Teams without having an on-prem option because of regulations and they start being under pressure.

Having a Skype for Business bridge might be a key element pushing them regarding Matrix usage as well as making it possible to even promote a migrations. Ignoring this may lead to loose support options / donations, patches... you name it. I think it makes sense to give another couple of big corporations or public authorities arguments to look at Matrix. Instead of letting them beg MS to get an self-hosted Teams.

So yes: Skype for Business is dead, MS Teams is the future if you stay in the Microsoft ecosystem. But if someone writes a bridge for it, it could help existing Skype for Business to convince migration to Matrix-based solutions as an option. And Skype for Business will still be there in the foreseeable future because there are large-scale on-prem installs out there under support which are not able or willing to migrate to Teams.
